engine/src/main/java/io/camunda/zeebe/engine/state/message/DbMessageSubscriptionState.java

```diff
@@ -250,7 +253,16 @@ public void visitPending(final long deadline, final MessageSubscriptionVisitor v
               pendingSubscription.elementInstanceKey(),
               BufferUtil.wrapString(pendingSubscription.messageName()));
 
-      visitor.visit(subscription);
+      if (subscription == null) {
+        // This case can occur while a scheduled job is running asynchronously
+        // and the stream processor removes one of the returned subscriptions from the state.
+        LOG.warn(
+            "Expected to find a subscription with key {} and message name {}, but none found. The state is inconsistent.",
+            pendingSubscription.elementInstanceKey(),
+            pendingSubscription.messageName());
+      } else {
```

Comment:
🔧 you can remove the else state by returning at the end of the if state

Reply:
@nicpuppa you probably mean using `continue` here. Because if I use `return` statement, it will go out of the loop and we will skip other subscriptions. Since, using `continue` is not a good practice of clean code, I prefer `else` statement instead. WDYT?

Reply:
Didn't saw it was a loop 👀 👓 , sorry for that! 

In this particular case using `continue` statement is not so bad, but is just my own preference. You can ignore my suggestion